testsuite/tests/basic-modules/recursive_module_evaluation_errors.ml

```diff
@@ -74,11 +74,11 @@ Error: Cannot safely evaluate the definition of the following cycle
 Line 2, characters 2-41:
 2 |   module F: functor(X:sig end) -> sig end
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-  Module "A" defines an unsafe functor, "F" .
+  Module "A" defines an unsafe functor, "A.F.F" .
```

Comment:
Something went wrong here, should this be `A.F`?

Reply:
I've made a change to account for when the subid would be the last module encountered so we don't double up. Let me know if the test cases are still not intact. Thank you!